contracts/swaps/SwapsUser.sol

```diff
@@ -99,7 +100,8 @@ contract SwapsUser is State, SwapsEvents, FeesHelper {
 		uint256[3] memory vals,
```

Comment:
consider using structs instead of arrays to reduce argument number. 
iirc it should work at least for internal functions (I'm not sure if 0.5.17 compiler allows it)